mux_test.go

```diff
@@ -1123,6 +1123,30 @@ func TestWalkNested(t *testing.T) {
 	}
 }
 
+func TestSubrouterErrorHandling(t *testing.T) {
+	superRouterCalled := false
+	subRouterCalled := false
+
+	router := NewRouter()
+	router.NotFoundHandler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		superRouterCalled = true
+	})
+	subRouter := router.PathPrefix("/bign8").Subrouter()
```

Comment:
Can remove the self reference if desired.  Just figured I would have a little fun with the test. :wink: